src/wallet/rpcwallet.cpp

```diff
@@ -1575,9 +1574,9 @@ static UniValue listsinceblock(const JSONRPCRequest& request)
     // when a reorg'd block is requested, we also list any relevant transactions
     // in the blocks of the chain that was detached
     UniValue removed(UniValue::VARR);
-    while (include_removed && paltindex && paltindex != pindex) {
+    while (include_removed && altheight && *altheight > height.value_or(-1)) {
         CBlock block;
-        if (!ReadBlockFromDisk(block, paltindex, Params().GetConsensus())) {
+        if (!pwallet->chain().findBlock(blockId, &block) || block.IsNull()) {
```

Comment:
in commit c4b750f:

Previously this would take an argument that was modified in each loop run, now the arguments are constant. That doesn't seem right.